r/R/parquet.R

```diff
@@ -373,6 +380,9 @@ ParquetFileWriter$create <- function(schema,
                                      sink,
                                      properties = ParquetWriterProperties$create(),
                                      arrow_properties = ParquetArrowWriterProperties$create()) {
+  if (!inherits(sink, "OutputStream")) {
+    abort("ParquetFileWriter sink must be an FileOutputStream or BufferOutputStream")
+  }
```

Comment:
We have a helper for this kind of assertion: https://github.com/apache/arrow/blob/master/r/R/util.R#L33

```suggestion
  assert_is(sink, "OutputStream")
```